src/Observable.ts

```diff
@@ -79,12 +80,12 @@ export function toTask<A>(o: Observable<A>): Task<A> {
 /**
  * @since 0.6.0
  */
-export const observable: Monad1<URI> & Alternative1<URI> & Filterable1<URI> & MonadObservable1<URI> = {
+export const observable: Monad1<URI> & Alt1<URI> & Plus1<URI> & Filterable1<URI> & MonadObservable1<URI> = {
   URI,
   map: (fa, f) => fa.pipe(rxMap(f)),
   of,
   ap: (fab, fa) => combineLatest([fab, fa]).pipe(rxMap(([f, a]) => f(a))),
-  chain: (fa, f) => fa.pipe(mergeMap(f)),
+  chain: (fa, f) => fa.pipe(switchMap(f)),
```

Comment:
Theoretical we could replace `switchMap` with `exhaustMap` and would also get the memory problem solved and hold all laws right?

In practice you are totally right that `switchMap` should be the default `chain` should we add for completeness alternative `Monad` instances like `mergeMonad` and `exhaustMonad`?

Reply:
AFAIK `exhaustMap` waits for passed `Observable` to complete before subscribing to the new created one (result of `chain`). Looks like an incorrect behavior and I can't think of a use case for that.

Replacing `switchMap` with `mergeMap` is a matter of `const m = {...observable, chain: (fa, f) => fa.pipe(mergeMap(f))}` - is it really necessary to put it into the lib? What about other possible operator replacements? IMO `switchMap` covers 99% usecases.

Reply:
> AFAIK exhaustMap waits for passed Observable to complete before subscribing to the new created one (result of chain). Looks like an incorrect behavior and I can't think of a use case for that.

Just a theorical question sure for the use cases we are looking at it does different things.

> Replacing switchMap with mergeMap is a matter of const m = {...observable, chain: (fa, f) => fa.pipe(mergeMap(f))} - is it really necessary to put it into the lib? What about other possible operator replacements? IMO switchMap covers 99% usecases.

For the `ReaderObservable` this approach would not be practical and it would follow the implementation style of `fp-ts` like done for [taskSeq](https://github.com/gcanti/fp-ts/blob/master/src/Task.ts#L134). But of cause it could be part of another PR. Totally agree with the fact that `switchMap` covers most of the usecases.

Reply:
> follow the implementation style of fp-ts like done for taskSeq

Do you mean that there should be additional `mergeReaderObservable` and the same for any other instance using `Observable` as base?

Reply:
Thats what I would like to discuss to keep up with the `fp-ts` coding style. @gcanti any opinions?

Reply:
Honestly I don't understand the whole thing, keep in mind that I haven't used rxjs in at least two years. 

So here's some questions:

- is `switchMap` a drop in replacement of `mergeMap`? Or this is a breaking change?
- why this PR contains parts of #29? Is replacing `mergeMap` with `switchMap` somehow related?
- why `filterMap`'s implementation is changed?

Reply:
- it's a breaking change because it changes the behavior of the observable, and this break is desirable
- it contains tests from #29. So we wait for #29, then merge this
- new implementation is more efficient, previously a lot of unnecessary intermediate single-value observable were created

Reply:
So what is the decision, @gcanti?

Reply:
As I said above I got away from rxjs so if you (@raveclassic @mlegenhausen) both think this is a good change, I trust your judgment.

However this is a blocker:

> why this PR contains parts of #29?

Reply:
Then I would go for it.

Reply:
@raveclassic a question: I was looking at the following test

```ts
const failure = {
  fa: '                             ------------a----------------|',
  fab: '                            ---f-------------------------|',
  gac: '                            --------g--------------------|',

  'LEFT SIDE': '',
  'alt(fab, gac)': '                ---f----g--------------------|',
  'ap(alt(fab, gac), fa)': '        ------------c----------------|',

  'RIGHT SIDE': '',
  'ap(fab, fa)': '                  ------------b----------------|',
  'ap(gac, fa)': '                  ----------------c------------|',
  'alt(ap(fab, fa), ap(gac, fa))': '------------b---c------------|'
}
```

in the `RIGHT SIDE`

```ts
'alt(ap(fab, fa), ap(gac, fa))': '------------b---c------------|'
```

`b` and `c` are emitted in the same tick? Something like

```ts
'alt(ap(fab, fa), ap(gac, fa))': '------------x----------------|'
```

where `x = b and c`

Reply:
That's tricky :) There's no way to test `.toNotBe` on `TestScheduler`'s `expectObservable`, so we need to pass `assert.notDeepStrictEqual` as equality assertion function. Then we intentionally test that resulting observable _does not equal_ marbles, because they are "incorrect" but should be the same as in `success`' `RIGHT SIDE`.

Reply:
Waaaaaaaait... Let me recheck o_O

Reply:
Yeah, you are right, there's an error in the test. How could I miss it?... Anyway I updated the test and it still shows distributivity does not hold

Reply:
I have a proposal: for what I can undestand switching to `switchMap` is important to you, while a discussion about `Alt` / `Alternative` instances, albeit being very interesting from a theoretical point of view, is way less important in practice (it's just `merge` in the end), and currently is a blocker. 

So what about:

- replace `mergeMap` with `switchMap` (breaking change)
- optimize `filterMap` implementation
- **remove** `Alt` / `Alternative` instances from `Observable`, `ObservableEither` and `ReaderObservable` (it's a breaking change but we are already doing a breaking change for `switchMap`)
- release v0.7
- (maybe) continue the debate about Alternative in the future with less rush

@raveclassic @mlegenhausen what do you think?

Reply:
@gcanti We can keep `zero` as part of `Plus` typeclass, it doesn't require `alt` to be `distributive`. So that `Observable` is a `Plus & Alt`, but not an `Alternative`. This is not a breaking change, everything stays structurally equal.
But I'm open to any decision.